### PR TITLE
Create VM, RS Vault and configure backup

### PR DESCRIPTION
**Pull request by nilaydshah**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Creates simple windows virtual machine
* Creates recovery services vault
* Configures backup for virtual machine

### CODE REVIEW

**Comment by nilaydshah** [comment]
@bmoore-msft  can you please review this?

**Comment by nilaydshah** [comment]
@bmoore-msft can you please review/merge this PR?

File: 101-recovery-services-create-vm-and-configure-backup/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Create simple windows VM and configure backup
```
Comment (bmoore-msft) [inline 101-recovery-services-create-vm-and-configure-backup/README.md]: Create a simple...

Fix grammar throughout this file

File: 101-recovery-services-create-vm-and-configure-backup/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Create simple windows VM and configure backup
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-recovery-services-create-vm-and-configure-backup%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+<a href="http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-recovery-services-create-vm-and-configure-backup%2Fazuredeploy.json" target="_blank">
+    <img src="http://armviz.io/visualizebutton.png"/>
+</a>
+
+This template allows you to deploy simple Windows VM and Recovery Services Vault. Also configures Protection with DefaultPolicy
```
Comment (bmoore-msft) [inline 101-recovery-services-create-vm-and-configure-backup/README.md]: This template allows you to deploy simple Windows VM and Recovery Services Vault configured with a default protection policy.

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    "dnsLabelPrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+      }
+    },
+    "windowsOSVersion": {
+      "type": "string",
+      "defaultValue": "2012-R2-Datacenter",
+      "allowedValues": [
```
Comment (bmoore-msft) [inline 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: why is 2016 not allowed?

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+        "2008-R2-SP1",
+        "2012-Datacenter",
+        "2012-R2-Datacenter"
+      ],
+      "metadata": {
+        "description": "The Windows version for the VM. This will pick a fully patched image of this given Windows version. Allowed values: 2008-R2-SP1, 2012-Datacenter, 2012-R2-Datacenter."
+      }
+    },
+    "vaultName": {
```
Comment (bmoore-msft) [inline 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: does this need to be globally unique?  If not, just move it to a variable

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+    "subnetPrefix": "10.0.0.0/24",
+    "storageAccountType": "Standard_LRS",
+    "publicIPAddressName": "myPublicIP",
+    "publicIPAddressType": "Dynamic",
+    "vmStorageAccountContainerName": "vhds",
+    "vmName": "BackedupWinVM",
+    "vmSize": "Standard_D1",
+    "virtualNetworkName": "MyVNET",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetID'), '/subnets/', variables('subnetName'))]",
```
Comment (bmoore-msft) [inline 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: simplify, just use resourceId() and remove the VnetID above

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+    "backupFabric": "Azure",
+    "backupPolicyName": "DefaultPolicy",
+    "protectionContainer": "[concat('iaasvmcontainer;iaasvmcontainerv2;', resourceGroup().name, ';', variables('vmName'))]",
```
Comment (bmoore-msft) [inline 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: are the naming requirements for a container compatible with those of resourceGroups?

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+              "subnet": {
+                "id": "[variables('subnetRef')]"
+              }
+            }
+          }
+        ]
+      }
+    },
+    {
+      "apiVersion": "2015-06-15",
```
Comment (bmoore-msft) [inline 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: update apiVersion

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json (hunk 4 above)
Comment (nilaydshah) [reply 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: Unique in subscription. We want user to give vault name as parameter, and hence I want to keep it parameter. Any pattern when to use variables vs parameter?

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json (hunk 6 above)
Comment (nilaydshah) [reply 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: Yes. by the way we have working template: https://github.com/Azure/azure-quickstart-templates/blob/master/101-recovery-services-backup-vms/azuredeploy.json

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json (hunk 7 above)
Comment (nilaydshah) [reply 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: Updated.

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json (hunk 5 above)
Comment (nilaydshah) [reply 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: Updated.

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json (hunk 4 above)
Comment (nilaydshah) [reply 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: Updated as per review comment. But just wanted to know when to use variable/parameter?

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json (hunk 3 above)
Comment (nilaydshah) [reply 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: Missed this. Updated.

File: 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json (hunk 4 above)
Comment (bmoore-msft) [reply 101-recovery-services-create-vm-and-configure-backup/azuredeploy.json]: Generally if it's an endpoint or the user needs to know it for some reason, parameter is good - and even then a defaultValue is helpful (generate a unique one)

**Pull request merged**